### PR TITLE
Pyro client from remote job can use cylc-suite-env

### PR DESCRIPTION
**Pull request by matthewrmshin**
A Pyro client invoked by a remote task job can now use host and port information from the `cylc-suite-env` file. This allows commands such as broadcast to work from a task job running on a remote host without a shared file system with the suite host.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark please review.

File: lib/cylc/network/pyro_base.py [hunk 1]
```diff
@@ -124,9 +124,36 @@ def _set_uri(self):
...
+                    elif key == 'CYLC_SUITE_PORT':
+                        port = value
+                if suite and host and port:
+                    # This ensures that all values are good
+                    uri_data['host'] = host
+                    uri_data['port'] = port
+                    self.host = host
+                    self.port = port
+            except (IOError, ValueError):
+                pass
```
Comment (hjoliver) [inline lib/cylc/network/pyro_base.py]: The `cylc-suite-env` file is also read and parsed in `lib/cylc/task_message`, but with different code.  Maybe time for a small module that handles parsing (and writing) the file, just returning a dict of parsed values?

**Comment by hjoliver** [comment]
Definitely fixes the problem. One comment above.

File: lib/cylc/network/pyro_base.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/cylc/network/pyro_base.py]: I did have a look at the logic, and I can have a look again. The write logic is done in 3 lines in `cylc.scheduler.Scheduler.configure`, (and the dict containing the data is defined in `cylc.scheduler.Scheduler.configure_suite_environment` in 7 lines). The read/parse logic in `cylc.task_message` is in 3 lines to override its own environment map, which is slightly different from the purpose here. I'll have a look to see what's the best way to combine everything.

**Comment by matthewrmshin** [comment]
@hjoliver This is now ready to be looked at again. Note that I have rebased the branch.

File: lib/cylc/suite_env.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    An instance of this class has the following attributes:
+
+    suite_name: the name of the suite
+    suite_host: the name of the host on which the suite is running
+    suite_port: the port number on host for connecting to the running suite
+    suite_owner: the owner user ID of the suite
+    suite_cylc_version: version of cylc for running the suite
+
+    Constansts:
```
Comment (hjoliver) [inline lib/cylc/suite_env.py]: spelling

**Comment by hjoliver** [comment]
Review 1 - good, test battery passes here; one trivial spelling error in a comment!

File: lib/cylc/suite_env.py (hunk 2 above)
Comment (matthewrmshin) [inline lib/cylc/suite_env.py]: Fixed.

**Comment by arjclark** [comment]
Review 2 - looks ok to me. Test battery behaving as expected.

**Pull request merged**